src/main/scala/tech/cryptonomic/conseil/db/DatabaseQueryExecution.scala

```diff
@@ -21,100 +21,100 @@ object DatabaseQueryExecution {
 
   sealed trait JoinedTables
 
-  case class BlocksOperationGroupsOperationsAccounts(
+  final case class BlocksOperationGroupsOperationsAccounts(
```

Comment:
What's the reasoning behind making all of these case classes final? What does this impact?

Reply:
You can't override them in any way using the `extends` mechanism, which is what you commonly want for defining "closed enumerations of types" as in ADTs.

More generally extending a case class usually breaks the methods automatically provided for you (e.g. `equals`, `hashcode`, `copy`), which are based on the original case class's fields for example

Look into this [SO answer](https://stackoverflow.com/questions/34561614/should-i-use-the-final-modifier-when-declaring-case-classes) for other details